osu.Game/Scoring/Legacy/LegacyScoreEncoder.cs

```diff
@@ -95,8 +95,9 @@ private string replayStringContent
 
                     foreach (var f in score.Replay.Frames.OfType<IConvertibleReplayFrame>().Select(f => f.ToLegacy(beatmap)))
                     {
-                        replayData.Append(FormattableString.Invariant($"{(int)Math.Round(f.Time - lastF.Time)}|{f.MouseX ?? 0}|{f.MouseY ?? 0}|{(int)f.ButtonState},"));
+                        replayData.Append(FormattableString.Invariant($"{Math.Round(f.Time - lastF.Time)}|{f.MouseX ?? 0}|{f.MouseY ?? 0}|{(int)f.ButtonState},"));
                         lastF = f;
+                        lastF.Time = Math.Round(f.Time);
```

Comment:
Instead of modifying `lastF` like this, using an `int` variable for the rounded previous time will express the meaning better (stable uses `int` for time variables, I guess?), in my opinion.
Also, please add a comment saying this is to match stable behavior.
And the expression `(int)Math.Round(f.Time - lastTime)` now can be simplfied to `Math.Round(f.Time) - lastTime`.

Reply:
I‘ve already removed the unnecessary casting.

Reply:
I mean:

```csharp
// some comment
int roundedLastTime = 0;
foreach (var f in ...) {
  int roundedTime = (int)Math.Round(f.Time); // or (int)f.Time? whichever matching stable.
  replayData.Append(... {roundedTime - lastRoundedTime} ...)
  lastRoundedTime = roundedTime;
}
```

Reply:
> And the expression (int)Math.Round(f.Time - lastTime) now can be simplfied to Math.Round(f.Time) - lastTime.

I don't think these two are equivalent? `round(9.8 - 1) != round(9.8) - 1`

Reply:
> I mean:
> 
> ```cs
> // some comment
> int roundedLastTime = 0;
> foreach (var f in ...) {
>   int roundedTime = (int)Math.Round(f.Time); // or (int)f.Time? whichever matching stable.
>   replayData.Append(... {roundedTime - lastRoundedTime} ...)
>   lastRoundedTime = roundedTime;
> }
> ```

``` C#
                    int lastTimeRounded = 0;
                    foreach (var f in score.Replay.Frames.OfType<IConvertibleReplayFrame>().Select(f => f.ToLegacy(beatmap)))
                    {
                        // Rounding because stable could only parse integral values
                        int timeRounded = (int)Math.Round(f.Time);
                        replayData.Append(FormattableString.Invariant($"{timeRounded - lastTimeRounded}|{f.MouseX ?? 0}|{f.MouseY ?? 0}|{(int)f.ButtonState},"));
                        lastF = f;
                        lastTimeRounded = timeRounded;
                    }
```
Is this good?

Reply:
`lastF` is no longer needed. Beside that, that is the code I imagined.